dd-sdk-android-compose/src/main/kotlin/com/datadog/android/compose/InteractionTracking.kt

```diff
@@ -55,3 +158,273 @@ internal class TapActionTracker(
         onTap.value.invoke()
     }
 }
+
+@Suppress("LongParameterList")
+internal suspend fun trackSwipe(
+    rumMonitor: RumMonitor,
+    targetName: String,
+    interactionSource: InteractionSource,
+    interactionType: InteractionType.Swipe<*>,
+    isRtl: Boolean,
+    attributes: Map<String, Any?>
+) {
+    trackDragInteraction<SwipeStartProps>(
+        interactionSource,
+        onStart = { interactions, start ->
+            interactions[start] = SwipeStartProps(
+                interactionType.swipeableState.currentValue,
+                // roundToInt can throw exception for Float.NaN, but we won't get such value
+                @Suppress("UnsafeThirdPartyFunctionCall")
+                interactionType.swipeableState.offset.value.roundToInt()
+            )
+            rumMonitor.startUserAction(RumActionType.SWIPE, targetName, emptyMap())
+        },
+        onStopOrCancel = { startProps ->
+            reportSwipeInteraction(
+                rumMonitor,
+                targetName,
+                startProps,
+                interactionType,
+                isRtl,
+                attributes
+            )
+        }
+    )
+}
+
+@Suppress("LongParameterList")
+internal suspend fun trackScroll(
+    rumMonitor: RumMonitor,
+    targetName: String,
+    interactionSource: InteractionSource,
+    interactionType: InteractionType.Scroll,
+    isRtl: Boolean,
+    attributes: Map<String, Any?>
+) {
+    trackDragInteraction<ScrollStartProps>(
+        interactionSource,
+        onStart = { interactions, start ->
+            interactions[start] =
+                ScrollStartProps(
+                    interactionType.scrollableState.currentPosition
+                )
+            rumMonitor.startUserAction(RumActionType.SCROLL, targetName, emptyMap())
+        },
+        onStopOrCancel = { startProps ->
+            reportScrollInteraction(
+                rumMonitor,
+                targetName,
+                startProps,
+                interactionType,
+                isRtl,
+                attributes
+            )
+        }
+    )
+}
+
+internal const val FROM_SWIPE_STATE_ATTRIBUTE = "from_state"
```

Comment:
Do we really need to report these values ?

Reply:
Good question. I'm still thinking about it. Initially my motivation was to give an opportunity to filter out swipe interactions which didn't change the state (if user wants) in the scrubbing API, especially considering we removed side-effect requirement.

On another hand https://issuetracker.google.com/issues/207695812 is resolved as `Intended behavior`, so we won't know correct target state at the end of interaction (although it may be an edge case - very fast and very strong swipe).

Reply:
I would say keep them for now then.